ams/ams-server/src/main/java/com/netease/arctic/ams/server/terminal/TerminalSessionContext.java

```diff
@@ -0,0 +1,305 @@
+  }
+
+  public List<StatementResult> getStatementResults() {
+    return task.get().executionResult.getResults();
+  }
+
+  public long lastExecutionTime() {
+    return lastExecutionTime;
+  }
+
+  public String lastScript() {
+    ExecutionTask t = this.task.get();
+    if (t == null) {
+      return "";
+    }
+    return t.script;
+  }
+
+  private synchronized TerminalSession lazyLoadSession() {
+    if (session == null) {
+      task.get().executionResult.appendLog("terminal session dose not exists. create session first");
+      session = factory.create(metaStore, sessionConfiguration);
+      task.get().executionResult.appendLog("create a new terminal session.");
+    }
+    return session;
+  }
+
+  private class ExecutionTask implements Supplier<ExecutionStatus> {
+
+    final String script;
+
+    final ExecutionResult executionResult = new ExecutionResult();
+
+    private final AtomicBoolean canceled = new AtomicBoolean(false);
+    private final int fetchLimits;
+    private final boolean stopOnError;
+    private final Supplier<TerminalSession> sessionSupplier;
+    private final String catalog;
+
+    public ExecutionTask(
+        String catalog,
+        String script,
+        int fetchLimits,
+        boolean stopOnError,
+        Supplier<TerminalSession> sessionSupplier) {
+      this.catalog = catalog;
+      this.script = script;
+      this.fetchLimits = fetchLimits;
+      this.stopOnError = stopOnError;
+      this.sessionSupplier = sessionSupplier;
+    }
+
+    @Override
+    public ExecutionStatus get() {
+      try {
+        return metaStore.doAs(() -> {
+          TerminalSession session = sessionSupplier.get();
+          executionResult.appendLog("fetch terminal session: " + sessionId);
+          executionResult.appendLogs(session.logs());
+          return execute(session);
+        });
+      } catch (Throwable t) {
+        LOG.error("something error when execute script. ", t);
+        executionResult.appendLog("something error when execute script.");
+        executionResult.appendLog(getStackTraceAsString(t));
+        return ExecutionStatus.Failed;
+      }
+    }
+
+    public void cancel() {
+      canceled.set(true);
+    }
+
+    ExecutionStatus execute(TerminalSession session) throws IOException {
+      LineNumberReader reader = new LineNumberReader(new StringReader(script));
+      StringBuilder statementBuilder = null;
+      String line;
+      int no = -1;
+
+      while ((line = reader.readLine()) != null) {
+        if (canceled.get()) {
+          executionResult.appendLog("execution is canceled. ");
+          return ExecutionStatus.Canceled;
+        }
+        if (statementBuilder == null) {
+          statementBuilder = new StringBuilder();
+        }
+        line = line.trim();
+        if (line.length() < 1 || line.startsWith("--")) {
+          // ignore blank lines and comments.
+          continue;
+        } else if (line.endsWith(";")) {
```

Comment:
It seems that now only one statement is allowed on a line，so maybe add some comments about that and a TODO for future improvement.